src/app/BUILD.gn

```diff
@@ -427,6 +427,7 @@ static_library("app") {
   output_name = "libCHIPDataModel"
 
   sources = [
+    "${chip_root}/src/controller/ExamplePersistentStorage.cpp",
```

Comment:
This must be a separate library that is NOT a dependency of libCHIPDataModel since it's common code shared by all. Specific examples have to be split AND injectable. Not baked-in.